File: typescript/src/platform.ts
import { SpongePlatform } from "@paysponge/sdk";
import { requireEnv, spongeBaseUrl } from "./env.js";

const platform = await SpongePlatform.connect({
  apiKey: requireEnv("SPONGE_MASTER_KEY"),
  baseUrl: spongeBaseUrl(),
});

const { agent, apiKey } = await platform.createAgent({
  name: `starter-agent-${Date.now()}`,
  description: "Created from the Sponge TypeScript starter pack",
  dailySpendingLimit: "10",
});

console.log("Created agent");
console.log({ id: agent.id, name: agent.name });

const wallet = await platform.connectAgent({
  apiKey,
  agentId: agent.id,
});

console.log("Agent addresses");
console.log(await wallet.getAddresses());

File: python/examples/platform.py
import time

from paysponge import SpongePlatform

from _env import load_dotenv, require_env, sponge_base_url

load_dotenv()

platform = SpongePlatform.connect(
    api_key=require_env("SPONGE_MASTER_KEY"),
    base_url=sponge_base_url(),
)

agent = platform.create_agent(
    name=f"starter-agent-{int(time.time())}",
    description="Created from the Sponge Python starter pack",
    daily_spending_limit="10",
)

print("Created agent")
print({"id": agent.id, "name": getattr(agent, "name", None)})
